### ISSUE TITLE
Keep track of users' block/unblock actions for moderation purposes

### ISSUE THREAD

**Issue by SoniEx2** [tootsuite/mastodon#14965]
### Pitch

Mastodon should keep a log of users' block/unblock actions so that moderators can look at when a block was placed, when a post was made, and when a block was removed, and so on.

### Motivation

We once tried to forgive an abuser and our admins had no way of checking that they were trying to send us toots while we had them blocked. So our admins had no way to confirm that they were actually an abuser. This would solve that.

**Comment by test2a** [comment]
Yes. Reddit has a similar thing where once in the moderation log, if a user is reported, all their recent posts and interactions with mods, reports, results and such are on the sidebar. This gives the mod more info about the user because users tend to not reveal stuff on their own.

**Comment by ThisIsMissEm** [comment]
This could significantly increase the amount of data in the audit log; so that's something we need to be conscious of when considering something like this. Currently the audit log is mostly about actions which moderators / admins have done.

We could add this but I'd be wary of the additional storage cost.
